### DISCUSSION TITLE
Reprograming Data and CellRank

### DISCUSSION THREAD

**Discussion by AP-596** [theislab/cellrank#1179] [Biological applications]
Hello everyone, 

I have a reprogramming dataset that I would very much like to apply cellrank on! I was very intrigued by the fact that CellRank could predicted the reprogramming outcome. I have mainly looked at the [experimental time tutorial ](https://cellrank.readthedocs.io/en/stable/notebooks/tutorials/kernels/500_real_time.html) and until here everything is great. I would like to translate the recovered dynamics (the result of this): 
`tmk.plot_random_walks(
    max_iter=500,
    start_ixs={"day": 0.0},
    basis="force_directed",
    seed=0,
    dpi=150,
    size=30,
)`
into trajectories or to put in a better way How can I now more about the each start and finish points? I am also utilizing the RNA velocity tutorial but I am a little bit confused as to how to move forward. 

Any ideas would be appreciated !

**Comment by WeilerP** [comment]
@AP-596, our _RealTimeKernel_-based analyses from CellRank 2 are probably what you are looking for: We applied the _RealTimeKernel_ to a dataset of [mouse embryonic fibroblasts](https://theislab.github.io/cellrank2_reproducibility/realtime_kernel/mef/README.html) and [pharyngeal endoderm development](https://theislab.github.io/cellrank2_reproducibility/realtime_kernel/pharyngeal_endoderm/README.html).

In short, you can use our estimator to identify terminal states and gene expression trends towards them.

**Reply by AP-596** [reply]
Thank you ! Actually what I am trying to get is something like PAGA through  the results of tmk.plot_single_flow() and tmk.plot_random_walks() . Basically I want to connect the clusters in a way to say that a goes to b and b goes to c etc on my force directed layout. 

Naively for every cluster, I have generated tmk.plot_single_flow()  plots to see which cluster is going to which cluster but I want this on the force directed layout. The velocity was fine, I combined the kernels from realtime and velocity (combined_kernel = 0.8 * vk + 0.2 * tmk) and velocity based on experimental time made sense. Up until this point, I have utilized [ExperimentalTime](https://cellrank.readthedocs.io/en/latest/notebooks/tutorials/kernels/500_real_time.html), [RNAVelocity tutorials](https://cellrank.readthedocs.io/en/latest/notebooks/tutorials/kernels/200_rna_velocity.html). 

I then followed the suggestion and utilized  cellrank2 realtime tutorial for MEFs with my own data and realtime pseudotime doesn't reflect the experimental time. It is mostly the same color and prevents me from having a conclusion about it. Then when I look at the initial and terminal states, my initial state is named as one of the terminals and the initial state is not the initial state. I am not the most python savvy person so sorry if this a naive question. 

I am sorry if this is too much mumbling but I just wanted connect the plots from  tmk.plot_single_flow() on my fdl in a way that would reflect the possible clusters going from one to another. Is there a way for me to do that? 

Also, this could be a little vague but I am not super comfortable sharing the plots as this a collaboration but if you have any questions please let me know. 

Thank you!

**Reply by Marius1311** [reply]
Hi @AP-596, currently, there's no functionality implemented in CellRank 1/2 to visualize discrete state transitions in a given embedding. You can visualize continuous fate probabilities in an embedding, and you can also aggregate fate probabilities to the cluster level the plot these as a PAGA plot. Please note that the original directed PAGA was designed to visualize a directed velocity graph and I haven't really experimented with using it for other kernels.

**Reply by AP-596** [reply]
Thank you very much !
